### PR TITLE
Filterbytag

### PR DESCRIPTION
**Pull request by omaretna**
add possibility to filter documentation by tag and custom tag.

exemple: apidoc --filter-by apigroup=MyGroup

### CODE REVIEW

**Comment by NicolasCARPi** [comment]
Hello,

Thank you for your contributions! I'll try and test that asap!

**Comment by omaretna** [comment]
Hello @NicolasCARPi,
Supper thanks.
Related to https://github.com/apidoc/apidoc/pull/810

File: lib/parser.js [hunk 1]
```diff
@@ -426,10 +435,24 @@ Parser.prototype._findBlockWithApiGetIndex = function(blocks) {
                 found = false;
                 break;
             }
+            // if (filterTag && blocks[i][j].name.substr(0, 10) === filterTag && blocks[i][j].content !== valueTofilter) {
```
Comment (NicolasCARPi) [inline lib/parser.js]: Please remove commented out code

Comment (omaretna) [reply lib/parser.js]: Done

**Comment by omaretna** [comment]
> Can you also add simple comments to your code to tell why you are doing things? What is the purpose of variables for instance, and what a block of code is supposed to achieve.
> 
> Also, I see you use `const` and `let instead of `var`, which is inconsistent with the rest of the codebase. But it's alright, it's fine to use it in 2019 and slowly the codebase should evolve towards it. :)

I used const and let but i change it to var to be consistent with the rest.
I add some comment i hope is fine.

File: lib/parser.js [hunk 2]
```diff
@@ -49,6 +50,12 @@ function Parser(_app) {
             self.addParser(parser, require(filename));
         }
     });
+
+    // check app.options.filterBy and define the tag to filer by
```
Comment (NicolasCARPi) [inline lib/parser.js]: typo here

File: lib/parser.js [hunk 3]
```diff
@@ -427,9 +438,26 @@ Parser.prototype._findBlockWithApiGetIndex = function(blocks) {
...
+                if (blocks[i][j].name.substr(0, filterTag.length) === filterTag && blocks[i][j].content === valueTofilter) {
+                    isToFilterBy = true;
+                }
+            }
+
             if (blocks[i][j].name.substr(0, 3) === 'api')
                 found = true;
         }
+
+        // add block if it's apidefine or the tag is equale to the value defined in options.
```
Comment (NicolasCARPi) [inline lib/parser.js]: "equal" and no dot at the end please

File: lib/parser.js [hunk 4]
```diff
@@ -427,9 +438,26 @@ Parser.prototype._findBlockWithApiGetIndex = function(blocks) {
                 break;
             }
 
+            // check if the user want to filter by some specifique tag
```
Comment (NicolasCARPi) [inline lib/parser.js]: "wants" and "specific" (are you french? :p)

Comment (omaretna) [reply lib/parser.js]: hh yes :p

File: lib/parser.js (hunk 2 above)
Comment (omaretna) [reply lib/parser.js]: Done

File: lib/parser.js (hunk 3 above)
Comment (omaretna) [reply lib/parser.js]: Done

File: lib/parser.js (hunk 4 above)
Comment (omaretna) [reply lib/parser.js]: Done

**Comment by NicolasCARPi** [comment]
ok thanks I just need to test it and I'll merge!

**Comment by omaretna** [comment]
> ok thanks I just need to test it and I'll merge!

Supe thanks :)

**Comment by NicolasCARPi** [comment]
Merged! Thanks again for your contribution. This repository is in need of contributors, so don't hesitate if you feel you can tackle other issues ;)

~Nico

**Comment by omaretna** [comment]
i will try thanks :)

**Pull request merged**